src/LibraryManager/LibrariesValidator.cs

```diff
@@ -24,141 +28,346 @@ internal class LibrariesValidator
         /// <param name="defaultProvider"></param>
         public LibrariesValidator(IDependencies dependencies, string defaultDestination, string defaultProvider)
         {
-            _dependencies = dependencies ?? throw new ArgumentNullException(nameof(dependencies));
+            _dependencies = dependencies;
             _defaultDestination = defaultDestination;
             _defaultProvider = defaultProvider;
         }
-
-        private string _defaultDestination;
-        private string _defaultProvider;
-        private IDependencies _dependencies;
-
+        
         /// <summary>
-        /// Returns a collection of ILibraryOperationResult that represents the status for validation of each 
-        ///  library 
+        /// Validates all properties of the library including detection of invalid files
         /// </summary>
-        /// <param name="libraries"></param>
+        /// <param name="library"></param>
         /// <param name="cancellationToken"></param>
         /// <returns></returns>
-        public async Task<IEnumerable<ILibraryOperationResult>> GetLibrariesErrorsAsync(
-            IEnumerable<ILibraryInstallationState> libraries,
-            CancellationToken cancellationToken)
+        public async Task<ILibraryOperationResult> ValidateLibraryAsync(ILibraryInstallationState library, CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
 
-            IEnumerable<ILibraryOperationResult> validateLibraries = ValidatePropertiesAsync(libraries, cancellationToken);
+            var list = new List<IError>();
 
-            if (!validateLibraries.All(t => t.Success))
+            // Validates provider
+            IError error;
+            IProvider provider = GetProvider(library, out error);
+            if (provider == null)
             {
-                return validateLibraries;
+                list.Add(error);
             }
+            else
+            {
+                // Validates libraryId
+                if (!IsValidLibraryId(library, provider, out error))
+                {
+                    list.Add(error);
+                }
+
+                // Validates destinationPath
+                if (!IsValidDestinationPath(library, out error))
+                {
+                    list.Add(error);
+                }
 
-            IEnumerable<ILibraryOperationResult> expandLibraries= await ExpandLibrariesAsync(libraries, cancellationToken);
-            if (!expandLibraries.All(t => t.Success))
+                // Validates list of files if specified
+                if (!provider.SupportsRenamming)
```

Comment:
>SupportsRenamming [](start = 30, length = 17)

Why does renaming matter here?  Shouldn't we still be validating the list of output files even if it's a rename?

Reply:
According to the original implementation, if the Files property has a file name != from the source, we still copy the contents to it. Please log a bug if you think this should change.

---
In reply to: [195516678](https://github.com/aspnet/LibraryManager/pull/164#discussion_r195516678) [](ancestors = 195516678)